### DISCUSSION TITLE
Combine storage driver

### DISCUSSION THREAD

**Discussion by SF73** [jupyterlite/jupyterlite#1449] [Q&A]
Is there a way to use different storage drivers for different paths?

I have noticed better performance using /tmp (inmemory FS of pyodide) compared to the "mounted" indexDB.
However, /tmp is not accessible from lab file browsers.

I'd like to have my notebooks in persistent storage and input/output files in memory but visible in the browser so I can use all the features of lab (markdown, pdf, csv preview).

Note that it's not a big deal, there are workarounds like working in memory and copying the files back at the end or using BytesIO etc...
I'm just trying to find a compromise between performance and usability. 
If you have any advice and suggestions for best practices, I'd be glad to hear them.

**Comment by jtpio** [comment]
Maybe this could be implemented as a separate JupyterLab drive. 

Currently the JupyterLite contents manager relies on IndexedDB to store files in the browser. There has been some exploratory work to move that part to a separate extension in https://github.com/jupyterlab-contrib/jupyterlab-browser-storage, to allow switching contents manager more easily.

Also the kernels currently hardcode the mountpoint to `/drive`, for example here: https://github.com/jupyterlite/pyodide-kernel/blob/f16cba8dcecd26c23361f8a28c708c4006812f6c/packages/pyodide-kernel/src/worker.ts#L134
But this could be made configurable.
